### PR TITLE
Bala wlan0 wip queue devel

### PR DESCRIPTION
**Pull request by balamurugana**

### CODE REVIEW

File: pkg/uevent/queue.go [hunk 1]
```diff
@@ -68,42 +69,25 @@ func newEventQueue() *eventQueue {
 //   Update()
```
Comment (wlan0) [inline pkg/uevent/queue.go]: we can remove these comments. They were used only while discussing on the video call

File: pkg/uevent/queue.go [hunk 2]
```diff
@@ -119,15 +103,44 @@ func (e *eventQueue) push(event *deviceEvent) {
...
-	} else {
-		e.keyCh <- event.devPath
 	}
+
 	e.events[event.devPath] = event
+	return existingEvent, found
+}
+
+func (e *eventQueue) push(event *deviceEvent) {
+	existingEvent, found := e.set(event)
```
Comment (wlan0) [inline pkg/uevent/queue.go]: tabwidth should be 4 spaces across the project. Everyone should standardize on this. For now, please set it to 4 spaces. We can add a separate PR to add a linter to check for this

File: pkg/uevent/queue.go [hunk 3]
```diff
@@ -68,42 +69,25 @@ func newEventQueue() *eventQueue {
...
+
+func (e *eventQueue) set(event *deviceEvent) (existingEvent *deviceEvent, found bool) {
 	e.mutex.Lock()
 	defer e.mutex.Unlock()
 
-	addBackoff = event.popped
-
-	existingEvent, found := e.events[event.devPath]
+	existingEvent, found = e.events[event.devPath]
 	if found {
```
Comment (Praveenrajmani) [inline pkg/uevent/queue.go:65]: The `else if` block will never be executed if the existing event is found with proper order. 

We can change to this,

```suggestion
	if found && existingEvent.created.After(event.created) {
	     // Skip as passed event is older.
	     return nil, found
        }
```

File: pkg/uevent/queue.go [hunk 4]
```diff
@@ -119,15 +103,44 @@ func (e *eventQueue) push(event *deviceEvent) {
...
+	if !event.popped {
+		if existingEvent.timer != nil {
+			if existingEvent.timer.Stop() {
+				// As incoming event is new and existing event has a timer and
+				// we were able to stop the timer, send key to keyCh.
+				e.keyCh <- event.devPath
+			}
+			// As we are not able to stop the timer i.e. the timer already elapsed,
+			// the callback function in the timer sends key to keyCh.
+		}
```
Comment (Praveenrajmani) [inline pkg/uevent/queue.go]: AFAICS, this code block will never be executed as the events that aren't popped yet will never have timer set. (ie, existingEvent.timer will be nil always)

we can just have,

```
if existingEvent.timer != nil {
	if existingEvent.timer.Stop() {
		e.keyCh <- event.devPath
	}
}
```

File: pkg/uevent/queue.go (hunk 3 above)
Comment (balamurugana) [reply pkg/uevent/queue.go:65]: If we do that, we miss adding newer events for the same key.

File: pkg/uevent/queue.go (hunk 4 above)
Comment (balamurugana) [reply pkg/uevent/queue.go]: If passed event was updated in the map for same key and the event is new i.e. not popped, we don't need to send the key to `keyCh`

**Pull request merged**